src/main/java/com/iota/iri/Iota.java

```diff
@@ -106,7 +106,7 @@ public Iota(IotaConfig configuration) throws IOException {
         replicator = new Replicator(node, configuration);
         udpReceiver = new UDPReceiver(node, configuration);
         ledgerValidator = new LedgerValidator(tangle, milestoneTracker, transactionRequester, messageQ);
-        tipsSolidifier = new TipsSolidifier(tangle, transactionValidator, tipsViewModel);
+        tipsSolidifier = new TipsSolidifier(tangle, transactionValidator, tipsViewModel, configuration);
```

Comment:
I thought about it, but that would mean we would check at 3 different places in Iota.java for config enabled/disabled. If we want to do that for more classes in the future, it will be ugly/unreadable.

For memory, the 3 instances passed are created anyways, only the logger and the class itself gets loaded. I dont know numbers but i think thats a discardable amount of ram.

Reply:
My thinking was that later on we will move to a nicer design using dependency injections (I resisted this in the beginning but Hans convinced me). In this new design we will not create objects needlessly (hopefully :grimacing: ). So I wanted the logic to be like what we want to achieve.

I already told Hans to do the same on his classes.

I see and understand your point but still, let's get the logic correct